### ISSUE TITLE
Incorrect term in _LRScheduler.

### ISSUE THREAD

**Issue by BelBES** [pytorch/pytorch#9560]
So, _LRScheduler use 'epoch' (last_epoch etc.) as a term which describe step of learning rate. But epoch is an artifical term, more commonly we operate in terms of steps. And I think that rename of 'last_epoch' to 'last_step' will be more correct.

**Comment by themightyoarfish** [comment]
This is really … questionable, for lack of a better term. Considering `epoch` is _always_ used referring to iterating over the training set once, it is incredibly surprising to see the LR schedulers use it as the training step number (batch counter).

**Comment by ezyang** [comment]
@themightyoarfish Yes, please do!

**Comment by lkhphuc** [comment]
Was going to propose the same thing in LR scheduler.
When I define an LR scheduler, I defined the shape of the LR over time steps.
When I call .step(), it should step one step on the LR curve I defined above.
When I call .step() should not matter, whether it is for every epoch, or for every iterations, or for every 10 iterations, etc, it depend only on me and the LR scheduler algorithm I'm following.
